tree/dataframe/inc/ROOT/RDF/ActionHelpers.hxx

```diff
@@ -1252,6 +1252,7 @@ class SnapshotHelper : public RActionImpl<SnapshotHelper<ColTypes...>> {
    BoolArrayMap fBoolArrays; // Storage for C arrays of bools to be written out
    std::vector<TBranch *> fBranches;     // Addresses of branches in output, non-null only for the ones holding C arrays
    std::vector<void *> fBranchAddresses; // Addresses associated to output branches, non-null only for the ones holding C arrays
+   Long64_t fMaxTreeSize{TTree::GetMaxTreeSize()}; // Revert when #6640 will be solved
```

Comment:
This leaves the world in a bad state in case an exception is thrown during the event loop (which happens 
Can you please use instead a RAII object instantiated at the beginning of RLoopManager::Run? Something like

```cpp
struct MaxTreeSizeRAII {
   Long64_t fOldMaxTreeSize;
   MaxTreeSizeRAII() : fOldMaxTreeSize(TTree::GetMaxTreeSize()) {
      TTree::SetMaxTreeSize(std::numeric_limits<Long64_t>::max());
   }
   ~MaxTreeSizeRAII() {
      TTree::SetMaxTreeSize(fOldMaxTreeSize);
   }
};

Reply:
Yes this makse totally sense, thank you!